### PR TITLE
Update command-line tests

### LINKED ISSUE #5640: [Tests] clean up commandline tests
The auxiliary files for the commandline tests should all be moved such that we have one directory per test, each containing multiple files. Also, the json and non-json tests could be combined to share the same form.

### PR DESCRIPTION
**Pull request by androlo**
### Description

Put tests inside folders, divided them up into standard_json and regular solidity tests, and modified the script.

Depends on https://github.com/ethereum/solidity/pull/5709 and https://github.com/ethereum/solidity/pull/5710
Fixes #5640

### CODE REVIEW

**Comment by chriseth** [comment]
looks good, thanks a lot! If you don't mind, I would like to rebase and split the commits a little.

**Comment by androlo** [comment]
> looks good, thanks a lot! If you don't mind, I would like to rebase and split the commits a little.

@chriseth either do that or let me know what is wrong and i can fix. i intend on making more commits on other issues so it's nice to know what's wrong.

**Comment by chriseth** [comment]
I force-pushed to your branch already - could you please review? It was just that there were too many modifications in a single commit, so it was a little hard to review.

**Comment by androlo** [comment]
Not entirely sure what I'm looking at tbh, so i understand what you mean when you say it's difficult to review. 

It seems like the folders are now directly under cmdlineTests instead of two separate folders, and the standard_json tests now has `standard_` in the beginning of the folder name to distinguish them from the regular sol tests. Also the script has been modified to detect if it's standard json or not.

If that's right then I guess it should be fine.

**Comment by androlo** [comment]
Really all i did was change the file names and move into folders, then updated the related script (the two sub scripts that run the tests). all the rest was just moving stuff around in the script, which didn't change anything but the order in which the tests were run.

If all you did was change folders and merged those two script sections into one then it is fine.

**Comment by chriseth** [comment]
Yeah exactly. The problem was that "moving stuff around in the script" is very hard to review, so I singled that out into its own commit. The biggest change relative to yours is that I removed the additional subdirectories. Not sure if that's much better, though.

**Comment by androlo** [comment]
@chriseth Ah. Ok sloppy PR by me then.. When I do other ones I will be more careful about making changes, and divide into more commits. Either way it should be fine.

File: test/cmdlineTests/standard_wrong_key_auxiliary_input.json [hunk 1]
```diff
@@ -1,14 +0,0 @@
...
-	"sources":
-	{
-		"A":
-		{
-			"content": "pragma solidity >=0.0; contract C { function f() public pure {} }"
-		}
-	},
-	"auxiliaryInput":
-	{
-		"key1": "test"
```
Comment (ekpyron) [inline test/cmdlineTests/standard_wrong_key_auxiliary_input.json]: It's a bit hard to see, but I think this one was replaced by a copy of ``standard_wrong_type_auxiliary_input`` by accident. I think it needs to be restored.

File: test/cmdlineTests/gas_test_dispatch_optimize/input.sol [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (ekpyron) [inline test/cmdlineTests/gas_test_dispatch_optimize/input.sol]: I guess we don't need to introduce this new line here.

**Review by ekpyron — CHANGES_REQUESTED** [review]
Most of it looks fine, but I think one test needs fixing, see comment below.

**Comment by chriseth** [comment]
@ekpyron should be fixed now.

**Review by ekpyron — APPROVED** [review]

**Pull request merged**